WordPress/Classes/ViewRelated/Blog/PublicizeConnectionURLMatcher.swift

```diff
@@ -0,0 +1,159 @@
+                return nil
+            }
+        }
+
+        // In a handful of cases, we're just looking for a substring or prefix in the URL
+        fileprivate var matchString: String? {
+            switch self {
+            case .declinePath:
+                return "/decline"
+            case .authorizationPrefix:
+                return "https://public-api.wordpress.com/connect/"
+            default:
+                return nil
+            }
+        }
+    }
+
+    /// @return True if the url matches the current authorization component
+    ///
+    static func url(_ url: URL, contains matchComponent: MatchComponent) -> Bool {
+        if let queryItem = matchComponent.queryItem {
+            return self.url(url, contains: queryItem)
+        }
+
+        if let matchString = matchComponent.matchString {
+            switch matchComponent {
+            case .declinePath:
+                return url.path.contains(matchString)
+            case .authorizationPrefix:
+                return url.absoluteString.hasPrefix(matchString)
+            default:
+                return url.absoluteString.contains(matchString)
+            }
+        }
+
+        return false
+    }
+
+    // Checks to see if the current QueryItem is present in the specified URL
+    private static func url(_ url: URL, contains queryItem: URLQueryItem) -> Bool {
+        guard let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: true)?.queryItems else {
+            return false
+        }
+
+        return queryItems.contains(where: { urlItem in
+            var result = urlItem.name == queryItem.name
+
+            if let value = queryItem.value {
+                result = result && (urlItem.value == value)
+            }
+
+            return result
+        })
+    }
+
+    // MARK: - Authorization Actions
+
+    /// Classify actions taken by the web API
+    ///
+    enum AuthorizeAction: Int {
+        case none
+        case unknown
+        case request
+        case verify
+        case deny
+    }
+
+    static func authorizeAction(for matchURL: URL) -> AuthorizeAction {
+        // Path oauth declines are handled by a redirect to a path.com URL, so check this first.
+        if url(matchURL, contains: .declinePath) {
+            return .deny
+        }
+
+        if !url(matchURL, contains: .authorizationPrefix) {
+            return .none
+        }
+
+        if url(matchURL, contains: .requestActionItem) {
+            return .request
+        }
+
+        // Check the rest of the various decline ranges
+        if url(matchURL, contains: .denyActionItem) {
+            return .deny
+        }
+
+        // LinkedIn
+        if url(matchURL, contains: .userRefused) {
+            return .deny
+        }
+
+        // Facebook and Google+
+        if url(matchURL, contains: .accessDenied) {
+            return .deny
+        }
+
+        // If we've made it this far and verifyRange is found then we're *probably*
```

Comment:
This `verifyRange` reference seems outdated, so maybe something like this would be more accurate:
```suggestion
        // If we've made it this far and the `action=verify` query param is present then we're *probably*
```

Reply:
👍